### PR TITLE
Integrate the dummy dataset with the learning UI

### LINKED ISSUE #37: Integrate the dataset interface with the learning UI

### PR DESCRIPTION
**Pull request by Nitwix**
Also some refactoring because my separation between view and presenter
didn't make much sense.

- [x] I will still need to integrate with #31 once it's merged.
- [x] Also needs a rebase from main (will do once #31 is merged)

### CODE REVIEW

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/LearningActivity.kt [hunk 1]
```diff
@@ -1,24 +1,125 @@
...
+    private val halfOpaque = opaque / 2
+    private val targetOnDragListener = View.OnDragListener { v, event ->
+        when (event.action) {
+            DragEvent.ACTION_DRAG_ENTERED -> dragEnteredCallback(v)
+            DragEvent.ACTION_DRAG_EXITED -> dragExitedCallback(v)
+            DragEvent.ACTION_DROP -> dropCallback(event, v)
+
+            // DragEvent.ACTION_DRAG_ENDED &
+            // DragEvent.ACTION_DRAG_LOCATION & DragEvent.ACTION_DRAG_STARTED
+            else -> true
```
Comment (NicolasVial) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/LearningActivity.kt:75]: What is the purpose of the commented lines here?

**Review by NicolasVial — APPROVED** [review]
Looks good to me. Thanks for TestUtils.kt :)

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/LearningActivity.kt (hunk 1 above)
Comment (Nitwix) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/LearningActivity.kt:75]: It's just to remember which cases go in the else branch, if I ever need to handle those cases in the future.

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivityTest.kt [hunk 2]
```diff
@@ -41,22 +31,6 @@ class AddPictureActivityTest {
...
-
-            override fun getDescription(): String {
-                return "Wait for $millis milliseconds"
-            }
-
-            override fun perform(uiController: UiController?, view: View?) {
-                uiController?.loopMainThreadForAtLeast(millis)
-            }
-        }
-
```
Comment (Jaakik) [inline app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivityTest.kt:59]: Maybe we put this function under Test/utils for example for refactoring purposes as most tests currently are using it

**Review by Jaakik — APPROVED** [review]

**Review by wattsecs — APPROVED** [review]

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivityTest.kt (hunk 2 above)
Comment (Nitwix) [reply app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivityTest.kt:59]: That's what I did x) The left column is how the code was before modification

**Pull request merged**